src/net/connect.hpp

```diff
@@ -68,10 +70,15 @@ void connect_base(std::string address, int port, Callback<bufferevent *> cb,
         return;
     }
 
-    // WARNING: set callbacks after connect() otherwise we free `bev` twice
-    // NOTE: In case of `new` failure we let the stack unwind
```

Comment:
I don't think we should remove these two comments. The former is useful to explain why the code structure is laid out like this and why other code structures would lead to memory errors.

The latter justifies the way in which we deal with out of memory.